### PR TITLE
Cluster test: networking and service scheduling

### PR DESCRIPTION
**Pull request by ndegory**
Test in the cluster/envoy image:
- docker version
  - checks that the Docker API is at least 1.29
- networking and service scheduling
  - creates an overlay network
  - creates a service on this network, with a placement constraint on nodes with label "amp.type.api==true" (server)
  - creates a service on same network, with a placement constraint on nodes with label "amp.type.core==true" (client)
  - the client connects to the server
  - the connection success rate is assessed

## Verification
- cd cluster/envoy && make build-image

Expected output:
```
$ docker run --rm -v /var/run/docker.sock:/var/run/docker.sock:ro appcelerator/amp-envoy check -h
run validation tests on the cluster

Usage:
  envoy check [flags]

Flags:
  -a, --all          all tests
      --scheduling   check Docker service scheduling
      --version      check Docker version

$ docker run --rm -v /var/run/docker.sock:/var/run/docker.sock:ro appcelerator/amp-envoy check -a
2017/06/29 16:23:05 Docker engine version 17.06.0-ce
2017/06/29 16:23:05 passed - api version = 1.3

2017/06/29 16:23:05 creating network amptest
2017/06/29 16:23:05 creating service check-server
2017/06/29 16:23:10 assertServiceHasRunningTasks(check-server) passed
2017/06/29 16:23:10 creating service check-client
2017/06/29 16:23:15 assertServiceHasRunningTasks(check-client) passed
2017/06/29 16:23:20 Counting request success rate
2017/06/29 16:23:20 removing service check-client (o6iq8w5fdgp60iwa79s3g0ay8)
2017/06/29 16:23:22 removing service check-server (8e9j4x1i26kuvzqvux6ozzspo)
2017/06/29 16:23:24 removing network amptest (mhlyx4yk75u3tgju2hjtneth5)
2017/06/29 16:23:24 passed - 904 connections / 904 success

$ docker run --rm -v /var/run/docker.sock:/var/run/docker.sock:ro appcelerator/amp-envoy check --version
2017/06/29 16:23:32 Docker engine version 17.06.0-ce
2017/06/29 16:23:32 passed - api version = 1.3

$ docker run --rm -v /var/run/docker.sock:/var/run/docker.sock:ro appcelerator/amp-envoy check --scheduling
2017/06/29 16:23:43 creating network amptest
2017/06/29 16:23:43 creating service check-server
2017/06/29 16:23:48 assertServiceHasRunningTasks(check-server) passed
2017/06/29 16:23:48 creating service check-client
2017/06/29 16:23:53 assertServiceHasRunningTasks(check-client) passed
2017/06/29 16:23:58 Counting request success rate
2017/06/29 16:23:58 removing service check-client (ikmpyxtusfv1cd5e25dx8jin4)
2017/06/29 16:24:00 removing service check-server (yenk1lj6f77tuan8cx9zmrq7a)
2017/06/29 16:24:02 removing network amptest (8s1v77tvicugeblyp9wqm69ut)
2017/06/29 16:24:02 passed - 1053 connections / 1053 success
```

### CODE REVIEW

**Review by JosephGJ — DISMISSED** [review]
LGTM

**Review by bquenin — COMMENTED** [review]
Not related to your PR but it might be a good idea to reference the `cluster` folder in the `fmt` rule in the Makefile.

File: cluster/envoy/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # =============================
 FROM golang:1.8-alpine as build
 # =============================
-RUN apk --no-cache add git
+RUN apk --no-cache add git gcc musl-dev
```
Comment (subfuzion) [inline cluster/envoy/Dockerfile]: For what it's worth, you can also just use [build-base](https://pkgs.alpinelinux.org/package/v3.3/main/x86/build-base) as a good general base package (equivalent to build-essential) when you want to ensure you have a good base toolchain without thinking about each specific build tool.

File: cluster/envoy/envoyChecks.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
+package envoy
```
Comment (subfuzion) [inline cluster/envoy/envoyChecks.go]: Just use lowercase for filenames, underscores as necessary, per Go conventions.

**Review by subfuzion — CHANGES_REQUESTED** [review]
@ndegory Looks good. I updated the makefile to keep it simple like other plugin. Please rename `envoyChecks.go` and then we're good to go. It won't be part of this PR, but we'll have to make some changes to what and how we print to stderr/stdout to be more CLI friendly.

**Review by subfuzion — APPROVED** [review]

**Pull request merged**